pkg/dataset/types.go

```diff
@@ -127,51 +136,56 @@ func (s *Set[T]) Remove(item T, r remediation.Remediation, id int64) {
 	}
 }
 
-func (s *PrefixSet) Contains(ip netip.Addr) remediation.Remediation {
+func (s *PrefixSet) Contains(ip netip.Addr) (remediation.Remediation, string) {
 	s.RLock()
 	defer s.RUnlock()
 	valueLog := s.logger.WithField("value", ip.String())
 	valueLog.Debug("checking value")
 	s.logger.Tracef("current items: %+v", s.Items)
 	r := remediation.Allow
+	prefixRemediation := remediation.Allow
+	origin := ""
 	for k, v := range s.Items {
 		if k.Contains(ip) {
-			prefixRemediation := v.GetRemediation()
+			prefixRemediation, origin = v.GetRemediationAndOrigin()
```

Comment:
Shouldn't the origin only update once that is the highest remediation? EG the same as `r = prefixRemediation`?